src/lib.rs

```diff
@@ -17,6 +17,38 @@ pub struct AutoInheritConf {
     pub prefer_simple_dotted: bool,
 }
 
+/// Rewrites a `path` dependency as being absolute, based on a given path
+fn rewrite_dep_paths_as_absolute<'a, P: AsRef<std::path::Path>>(
+    deps: impl Iterator<Item = &'a mut Dependency>,
+    parent: P,
+) {
+    deps.for_each(|dep| {
+        if let Dependency::Detailed(detail) = dep {
+            detail.path = detail.path.as_mut().map(|path| {
+                parent
+                    .as_ref()
+                    .join(path)
+                    .canonicalize()
+                    .unwrap()
+                    .to_string_lossy()
```

Comment:
Following up on the other comment: we should only fail if the _relative_ path segments are not UT8. It should not be a problem if the absolute canonicalized path contains non-UT8 segments, as long as those don't appear in the final manifest.